src/app/dynamic-forms/fields-wrapper/fields-wrapper.component.ts

```diff
@@ -1,29 +1,61 @@
-import {Component, EventEmitter, Input, OnInit, Output, ViewEncapsulation} from '@angular/core';
-import {AppStateService} from "../../services/app-state.service";
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output, ViewEncapsulation } from '@angular/core';
+import { AppStateService } from '../../services/app-state.service';
 
 @Component({
-  selector: 'app-fields-wrapper',
-  templateUrl: './fields-wrapper.component.html',
-  styleUrls: ['./fields-wrapper.component.scss'],
+  selector     : 'app-fields-wrapper',
+  templateUrl  : './fields-wrapper.component.html',
+  styleUrls    : ['./fields-wrapper.component.scss'],
   encapsulation: ViewEncapsulation.None
 })
-export class FieldsWrapperComponent implements OnInit {
+export class FieldsWrapperComponent implements OnInit, OnChanges {
   @Input() form;
   @Input() rowId;
   @Input() formSections;
   @Input() isChild: boolean;
   @Input() fieldsWithErrors;
   @Input() hasChildTableAccess: boolean;
+  @Input() formFieldMetaDatas: any = {};
   @Output() eventOccurred = new EventEmitter<any>();
   @Output() childFormOpened = new EventEmitter<any>();
   @Output() deleteDialogOpened = new EventEmitter<any>();
+  public showSpinner = false;
+
   constructor(private appStateService: AppStateService) { } // AppState service is outside of Dynamic forms
 
   ngOnInit(): void {
   }
 
   expansionClicked(section) {
+    if(this.form)
+    {
+      if(this.form.sections[1]){
```

Comment:
Not sure I understand this piece, is this hard coded to only work for the section at index 1?

Reply:
removed hard coded index

Reply:
this still looks like it's for section[1]

Reply:
removed hard coded index